dsalgo/sort.py

```diff
@@ -201,3 +205,120 @@ def partition(array, start, end):
 
     return high
 
+def countingSort(array, place):
+    size = len(array)
+    output = [0] * size
+    count = [0] * 10
+
+    # Calculate count of elements
+    for i in range(0, size):
+        index = array[i] // place
+        count[index % 10] += 1
+
+    # Calculate cummulative count
+    for i in range(1, 10):
+        count[i] += count[i - 1]
+
+    # Place the elements in sorted order
+    i = size - 1
+    while i >= 0:
+        index = array[i] // place
+        output[count[index % 10] - 1] = array[i]
+        count[index % 10] -= 1
+        i -= 1
+
+    for i in range(0, size):
+        array[i] = output[i]
+
+
+# Main function to implement radix sort
+def radixSort(array,reverse=False):
+    # Get maximum element
+    max_element = max(array)
+
+    # Apply counting sort to sort elements based on place value.
+    place = 1
+    while max_element // place > 0:
+        countingSort(array, place)
+        place *= 10
+    if reverse:
+         return array[::-1]
+    else:
+         return array
+
+def bucket_sort(arr,reverse=False):
+     ''' Bucket Sort
+        Complexity: O(n^2)
+        The complexity is dominated by nextSort
+     '''
+     # The number of buckets and make buckets
+     num_buckets = len(arr)
+     buckets = [[] for bucket in range(num_buckets)]
+     # Assign values into bucket_sort
+     for value in arr:
+         index = value * num_buckets // (max(arr) + 1)
+         buckets[index].append(value)
+    # Sort
+     sorted_list = []
+     for i in range(num_buckets):
+         sorted_list.extend(next_sort(buckets[i]))
+     if reverse:
+         return sorted_list[::-1]
+     else:    
+         return sorted_list
+
+def next_sort(arr):
+     # We will use insertion sort here.
+     for i in range(1, len(arr)):
+         j = i - 1
+         key = arr[i]
+         while arr[j] > key and j >= 0:
+             arr[j+1] = arr[j]
+             j = j - 1
+         arr[j + 1] = key
+     return arr
```

Comment:
@Aryamanz29 aryaman use flake 8 to test the pep8 guidelines.
you haven't followed pep8 guidelines properly

Reply:
okay